### PR TITLE
feat(web): RTCDataChannel backpressure helper

### PR DESCRIPTION
**Pull request by riatzukiza**
Introduce backpressure-aware sender utility for RTCDataChannel; use bufferedAmountLowThreshold and await 'bufferedamountlow' before sending bursts.

### CODE REVIEW

File: apps/duck-web/src/net/send.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const makeThrottledSender = (ch: RTGDataChannel) => {\n  ch.bufferedAmountLowThreshold = 1 << t ? 20 : 1 << 20; // 1 MiB\n  let pending: Array<() => void> = [];\n  ch.addEventListener('bufferedamountlow', () => {\n    const q = pending;\n    pending = [];\n    q.forEach(fn => fn());\n  });\n  const waitLow = () => new Promise<void>(resolve => {\n    if (ch.bufferedAmount <= ch.bufferedAmountLowThreshold) return resolve();\n    pending.push(resolve);\n  });\n  return async (chunk: ArrayBuffer) => {\n    if (ch.readyState !== 'open') return;\n    if (ch.bufferedAmount > ch.bufferedAmountLowThreshold) await waitLow();\n    ch.send(chunk);\n  };\n};\n
```
Comment (riatzukiza) [inline apps/duck-web/src/net/send.ts:1]: Typo + logic bug: `RTGDataChannel` → `RTCDataChannel`; `1 << t ? 20 : 1 << 20` uses undefined `t` and wrong precedence. Suggest:

```ts
export const makeThrottledSender = (ch: RTCDataChannel, threshold = 1 << 20) => {
  ch.bufferedAmountLowThreshold = threshold;
  let waiters: Array<() => void> = [];
  const flush = () => { const q = waiters; waiters = []; q.forEach(fn => fn()); };
  ch.addEventListener('bufferedamountlow', flush);
  const waitLow = () => new Promise<void>(resolve => {
    if (ch.bufferedAmount <= ch.bufferedAmountLowThreshold) return resolve();
    waiters = [...waiters, resolve];
  });
  return async (chunk: ArrayBuffer | Uint8Array) => {
    if (ch.readyState !== 'open') return;
    if (ch.bufferedAmount > ch.bufferedAmountLowThreshold) await waitLow();
    ch.send(chunk as any);
  };
};
```

Comment (riatzukiza) [inline apps/duck-web/src/net/send.ts:1]: Two blockers: (1) type is `RTCDataChannel`, not `RTGDataChannel`; (2) the ternary precedence is wrong and `t` is undefined. Suggested fix with configurable threshold and DI‑friendly design:

```ts
export const makeThrottledSender = (ch: RTCDataChannel, threshold = 1 << 20) => {
  ch.bufferedAmountLowThreshold = threshold;
  let waiters: Array<() => void> = [];
  const flush = () => { const q = waiters; waiters = []; q.forEach(fn => fn()); };
  ch.addEventListener('bufferedamountlow', flush);
  const waitLow = () => new Promise<void>(resolve => {
    if (ch.bufferedAmount <= ch.bufferedAmountLowThreshold) return resolve();
    waiters = [...waiters, resolve];
  });
  return async (chunk: ArrayBuffer | Uint8Array) => {
    if (ch.readyState !== 'open') return;
    if (ch.bufferedAmount > ch.bufferedAmountLowThreshold) await waitLow();
    ch.send(chunk as any);
  };
};
```

Also newline at EOF.

Comment (riatzukiza) [inline apps/duck-web/src/net/send.ts:1]: One-click fix with proper type + backpressure:

```suggestion
export const makeThrottledSender = (ch: RTCDataChannel, threshold = 1 << 20) => {
  ch.bufferedAmountLowThreshold = threshold;
  let waiters: Array<() => void> = [];
  const flush = () => { const q = waiters; waiters = []; q.forEach(fn => fn()); };
  ch.addEventListener('bufferedamountlow', flush);
  const waitLow = () => new Promise<void>(resolve => {
    if (ch.bufferedAmount <= ch.bufferedAmountLowThreshold) return resolve();
    waiters = [...waiters, resolve];
  });
  return async (chunk: ArrayBuffer | Uint8Array) => {
    if (ch.readyState !== 'open') return;
    if (ch.bufferedAmount > ch.bufferedAmountLowThreshold) await waitLow();
    ch.send(chunk as any);
  };
};
```

**Comment by riatzukiza** [comment]
@codex fix comments

File: apps/duck-web/src/net/send.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export const makeThrottledSender = (
+  ch: RTCDataChannel,
+  threshold = 1 << 20,
+) => {
+  ch.bufferedAmountLowThreshold = threshold;
+  let waiters: ReadonlyArray<() => void> = [];
```
Comment (riatzukiza) [inline apps/duck-web/src/net/send.ts:6]: @codex ❤️ the immutability. One micro-nit: `waiters` is `ReadonlyArray`, but we reassign to `[]` and spread+append later. That’s fine, but to avoid accidental mutation via reference, copy when flushing.

```suggestion
  let waiters: ReadonlyArray<() => void> = [];
  const flush = () => {
    const queue = [...waiters];
    waiters = [];
    queue.forEach((fn) => fn());
  };
```

Optional: remove the `as any` when sending by normalizing to `Uint8Array` via `chunk instanceof Uint8Array ? chunk : new Uint8Array(chunk)`.

**Comment by riatzukiza** [comment]
Checklist refresh (cc @codex):

- [x] Throttled sender factory + backpressure
- [x] `bufferedAmountLowThreshold` configurable (default 1 MiB)
- [ ] Normalize chunk to `Uint8Array` to drop `as any`
- [ ] Tests for threshold + bufferedamountlow edge cases
- [ ] Ensure event listener removed on channel close (follow-up ok)

Close the three and we’re green.

**Comment by riatzukiza** [comment]
Throttled sender doc + diagram: `docs/services/duck-web/THROTTLED_SENDER.md`.

@codex fixups to confirm: 
- type name `RTCDataChannel`, 
- set `bufferedAmountLowThreshold` (default ~1 MiB), 
- wait on `bufferedamountlow` before sending when over threshold, 
- ignore send if channel not open.

This is the web-side backpressure paired with the voice forwarder.
